### PR TITLE
Add aurora serverlessv2 support

### LINKED ISSUE #137: Support serverless v2
Have a question? Please checkout our [Slack Community](https://slack.cloudposse.com) or visit our [Slack Archive](https://archive.sweetops.com/). 

[Slack Community](https://slack.cloudposse.com)

## Describe the Feature

Add serverless v2 support

## Expected Behavior

Can create a serverless v2 cluster by this module

## Use Case

Create a serverless v2 cluster by this module

## Describe Ideal Solution

Add a new config section like [serverlessv2_scaling_configuration](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster#serverlessv2_scaling_configuration)

## Alternatives Considered

Create the cluster by AWS provider directly

## Additional Context

The cluster instance class is "db.serverless"

### PR DESCRIPTION
**Pull request by topikachu**
## what
* Add aurora serverlessv2 support

## why
* AWS releases aurora serverless v2.  
* Adopt this new feature in this rds cluster module

## references
*  closes #137
* [Terraform serverlessv2_scaling_configuration](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster#serverlessv2_scaling_configuration)

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -115,6 +115,16 @@ resource "aws_rds_cluster" "primary" {
...
 
+  dynamic "serverlessv2_scaling_configuration" {
+    for_each = var.serverlessv2_scaling_configuration[*]
+    content {
+      max_capacity = serverlessv2_scaling_configuration.value.max_capacity
+      min_capacity = serverlessv2_scaling_configuration.value.min_capacity
+    }
+  }
+
+
```
Comment (Gowiem) [inline main.tf:126]:
```suggestion
```

File: main.tf [hunk 2]
```diff
@@ -115,6 +115,16 @@ resource "aws_rds_cluster" "primary" {
...
   }
 
+  dynamic "serverlessv2_scaling_configuration" {
+    for_each = var.serverlessv2_scaling_configuration[*]
+    content {
+      max_capacity = serverlessv2_scaling_configuration.value.max_capacity
+      min_capacity = serverlessv2_scaling_configuration.value.min_capacity
+    }
+  }
+
```
Comment (Gowiem) [inline main.tf:125]:
```suggestion
```

**Comment by Gowiem** [comment]
@topikachu @kevcube this would require [AWS Provider v4.12.0](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#4120-april-28-2022)+. 

Will have to chat with the team about if we're ready to bump this module to the v4+ provider.

**Comment by topikachu** [comment]
> @topikachu @kevcube this would require [AWS Provider v4.12.0](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#4120-april-28-2022)+.
> 
> Will have to chat with the team about if we're ready to bump this module to the v4+ provider.

Thanks.
If it's difficult to upgrade to the new v4+, is it possible to publish a new major version (even beta) instead of replacing the old one?
Or at least to a new git branch so I can use the offically one instead of mine. ^-^

**Comment by Nuru** [comment]
@topikachu Would you please add a test for this? You can pretty much duplicate `examples/complete` and `test/src/examples_complete_test.go`, rename "complete" to "serverlessv2" and flip on the new option in the new test.

**Comment by alexjurkiewicz** [comment]
Hey @topikachu. Are you still working on this PR? I will take it over if not 🙌

**Comment by topikachu** [comment]
> Hey @topikachu. Are you still working on this PR? I will take it over if not 🙌

I'm here. Will add tests as you suggest.

**Comment by topikachu** [comment]
> Hey @topikachu. Are you still working on this PR? I will take it over if not 🙌

Done!
A lot to change because I have to upgrade AWS provider version and others.
Hope this doesn't break others too much.

Test by `make clean test`.

**Review by Nuru — CHANGES_REQUESTED** [review]
Don't worry about the test failures from the previous run.

File: examples/complete/main.tf [hunk 3]
```diff
@@ -4,21 +4,21 @@ provider "aws" {
...
-  cidr_block = "172.16.0.0/16"
+  ipv4_primary_cidr_block = "172.16.0.0/16"
 
   context = module.this.context
 }
 
 module "subnets" {
   source  = "cloudposse/dynamic-subnets/aws"
-  version = "0.28.0"
+  version = "2.0.2"
```
Comment (Nuru) [inline examples/complete/main.tf:16]: `dynamic-subnets` v2 requires Terraform version `>= 1.1.0`. Please update `versions.tf` accordingly

File: main.tf [hunk 4]
```diff
@@ -115,6 +115,14 @@ resource "aws_rds_cluster" "primary" {
     }
   }
 
+  dynamic "serverlessv2_scaling_configuration" {
```
[the remark this answers is not in the document]
Comment (joshmello) [reply main.tf:118]: Would setting the default variable value for `storage_encrypted` to `true` fix these issues?

[the remark this answers is not in the document]
Comment (topikachu) [reply main.tf:118]: > storage_encrypted

I believe this setting is caused by the old serverless configuration.
Shall we just work on the serverlessv2_scaling_configuration in this PR?

We can clean the old serverless later if required.

[the remark this answers is not in the document]
Comment (joshmello) [reply main.tf:118]: imo yes, they support encryption anyways @Nuru any thoughts?

**Review by Nuru — APPROVED** [review]
The  `bats` test failure is due to formatting differences among Terraform versions, and inconsistency in what the actions format with vs what the actions test with. Ignoring it for now.

Marking this version as `no-release` so we can release it as version 1.0, per Cloud Posse migration to production semantic versioning.

**Comment by topikachu** [comment]
@Nuru  I'm happy that this change is merged into the upstream.
I'd like to know if there's any timeline to release this version? So I can use the official cloudposse module in my project.

Thanks

**Comment by kevcube** [comment]
@topikachu it is released

https://github.com/cloudposse/terraform-aws-rds-cluster/releases/tag/1.0.0
